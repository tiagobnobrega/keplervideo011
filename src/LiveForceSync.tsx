import React, { useEffect, useState } from 'react';
import {
  ActivityIndicator,
  BackHandler,
  StyleSheet,
  Text,
  View,
} from 'react-native';
import {
  ingestChannelLineup,
  ingestProgramLineup,
} from './livetv/task/EpgSyncTask';
import { COLORS } from './styles/Colors';
import { scaleUxToDp } from './utils/pixelUtils';

const TIME_TO_LEAVE_APP = 3000;

const LiveForceSync = () => {
  // NOTE: You should replace the code with your own business logic to render the splash screen and show progress status.
  const [isInProgress, setIsInProgress] = useState<boolean>(true);

  // Progress of ingestion from 0-100.
  const [progress, setProgress] = useState<number>(0);

  const [failed, setFailed] = useState<boolean>(false);

  /**
   * Takes in a decimal representing the percentage of the progress made for channel ingestion and updates the progress bar.
   * For this example, we are treating channel ingestion completion as being 50% complete on the progress bar.
   * @param channelProgress - values between 0 and 1 to set current progress.
   */
  const channelProgressCallback = (channelProgress: number) => {
    // Channel ingestion is the first 50% of progress bar.
    setProgress(channelProgress * 50);
  };

  /**
   * Takes in a decimal representing the percentage of the progress made for program ingestion and updates the progress bar.
   * Move progress bar to 100% once program ingestion has completed.
   * @param programProgress - values between 0 and 1 to set current progress.
   */
  const programProgressCallback = (programProgress: number) => {
    // Program ingestion happens second and is the last 50% of progress bar.
    setProgress(50 + programProgress * 50);
  };

  const ingestLineup = async () => {
    try {
      await ingestChannelLineup(channelProgressCallback);
      await ingestProgramLineup(programProgressCallback);
      setIsInProgress(false);
      await new Promise(resolve => setTimeout(resolve, TIME_TO_LEAVE_APP));
      BackHandler.exitApp();
    } catch (error) {
      setIsInProgress(false);
      setFailed(true);
    }
  };

  useEffect(() => {
    /*
      NOTE: If the customer is not entitled to any channels, re-route to your app's login screen.
      Otherwise, call your ingestion logic to push your latest EPG data.
      You should add your own business logic to manage the user session before calling the ingestLineup function.
    */

    ingestLineup();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return (
    <View style={styles.container}>
      {isInProgress ? (
        <>
          <Text style={styles.title}>Updating Channel Info</Text>
          <Text style={styles.text}>
            This may take a few minutes, please do not leave this screen{' '}
          </Text>
          <ActivityIndicator style={styles.activityIndicator} />
          <View style={styles.progressBar}>
            <View
              style={{ backgroundColor: COLORS.BLUE, width: `${progress}%` }}
            />
          </View>
        </>
      ) : (
        <>
          {failed ? (
            <>
              <Text style={styles.title}>Channel Update Failed</Text>
              <Text style={styles.text}>Please try again later</Text>
            </>
          ) : (
            <>
              <Text style={styles.title}>Channel Sync Complete</Text>
              <Text style={styles.text}>This screen will exit shortly</Text>
            </>
          )}
        </>
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: COLORS.BLACK,
    justifyContent: 'center',
    alignItems: 'center',
  },
  title: {
    color: COLORS.WHITE,
    fontSize: scaleUxToDp(70),
    margin: scaleUxToDp(10),
  },
  text: {
    color: COLORS.WHITE,
    fontSize: scaleUxToDp(26),
    margin: scaleUxToDp(10),
  },
  progressBar: {
    height: scaleUxToDp(20),
    width: '50%',
    flexDirection: 'row',
    backgroundColor: COLORS.WHITE,
    borderColor: COLORS.BLACK,
    borderWidth: 2,
    borderRadius: scaleUxToDp(5),
  },
  activityIndicator: {
    padding: scaleUxToDp(20),
  },
});

export default LiveForceSync;
