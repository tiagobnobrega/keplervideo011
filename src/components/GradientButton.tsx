import { Button, ButtonVariant } from '@amzn/kepler-ui-components';
import LinearGradient from '@amzn/react-linear-gradient';
import React, { useState } from 'react';
import { StyleSheet, TextStyle, View, ViewStyle } from 'react-native';
import { COLORS } from '../styles/Colors';
import { scaleUxToDp } from '../utils/pixelUtils';

interface GradientButtonProps {
  label: string;
  onPress: () => void;
  colors: string[];
  buttonStyle?: ViewStyle;
  containerStyle?: ViewStyle;
  variant: ButtonVariant;
  labelStyle?: TextStyle;
  focusedStyle?: ViewStyle;
  borderRadius?: number;
  focusedBorderRadius?: number;
}

const GradientButton: React.FC<GradientButtonProps> = ({
  label,
  onPress,
  colors,
  buttonStyle,
  containerStyle,
  variant,
  labelStyle,
  focusedStyle,
  borderRadius = scaleUxToDp(5),
  focusedBorderRadius = scaleUxToDp(10),
}) => {
  const [isFocused, setIsFocused] = useState(false);
  let borderRadiusStyle = isFocused
    ? { borderRadius: focusedBorderRadius }
    : { borderRadius };

  const onFocus = () => {
    setIsFocused(true);
  };

  const onBlur = () => {
    setIsFocused(false);
  };

  return (
    <View
      style={[
        styles.buttonContainer,
        containerStyle,
        isFocused && [styles.focusedContainer, focusedStyle],
        borderRadiusStyle,
      ]}>
      <LinearGradient style={{ borderRadius }} colors={colors}>
        <Button
          label={label}
          style={[
            styles.button,
            buttonStyle,
            isFocused && styles.focusedButton,
            borderRadiusStyle,
          ]}
          variant={variant}
          onPress={onPress}
          onFocus={onFocus}
          onBlur={onBlur}
          labelStyle={[styles.label, labelStyle] as TextStyle}
        />
      </LinearGradient>
    </View>
  );
};

const styles = StyleSheet.create({
  buttonContainer: {
    borderWidth: 4,
    borderColor: 'transparent',
  },
  button: {
    backgroundColor: 'transparent',
  },
  label: {
    fontWeight: 'normal',
  },
  focusedContainer: {
    borderColor: COLORS.ORANGE,
  },
  focusedButton: {
    borderColor: 'transparent',
  },
});

export default GradientButton;
