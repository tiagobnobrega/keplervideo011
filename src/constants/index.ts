import { COLORS } from '../styles/Colors';

export enum DPADEventType {
  LEFT = 'left',
  RIGHT = 'right',
  UP = 'up',
  DOWN = 'down',
  PLAY = 'play',
  PAUSE = 'pause',
  PLAYPAUSE = 'playpause',
  SKIPBACKWARD = 'skip_backward',
  SKIPFORWARD = 'skip_forward',
  SELECT = 'select',
  MENU = 'menu',
  BACK = 'back',
  PAGE_UP = 'page_up',
  PAGE_DOWN = 'page_down',
  PAGE_LEFT = 'page_left',
  PAGE_RIGHT = 'page_right',
  INFO = 'info',
  MORE = 'more',
}

export const NetworkStatus = {
  CONNECTED: 'Connected',
  NOT_CONNECTED: 'Not Connected',
};

export const LoginStatus = {
  SIGNED_IN: true,
  SIGNED_OUT: false,
};

export const EVENT_KEY_DOWN = 0;
export const EVENT_KEY_UP = 1;

export const MAX_NARROW_SCREEN_WIDTH = 1300;

export const CAPTION_DISABLE_TEXT = 'Off';
export const CAPTION_DISABLE_ID = '-1';
export const L2_GRADIENT_COLORS = [
  `${COLORS.PALE_GRAY}99`,
  `${COLORS.MEDIUM_GRAY}99`,
  COLORS.CHARCOAL_GRAY,
];
export enum ReadyState {
  HAVE_NOTHING = 0,
  HAVE_METADATA = 1,
  HAVE_CURRENT_DATA = 2,
  HAVE_FUTURE_DATA = 3,
  HAVE_ENOUGH_DATA = 4,
}
