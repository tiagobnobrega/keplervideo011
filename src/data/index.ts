import { RotatorData } from '../components/rotator/type';
import { TitleData } from '../types/TitleData';

export const TestData: () => TitleData[] = () => require('./dynamicData.json');
export const CostaRicaAttractionsData: () => TitleData[] = () =>
  require('./costaRicaAttractions.json');
export const CostaRicaIslandsData: () => TitleData[] = () =>
  require('./costaRicaIslands.json');
export const CostaRicaUnderwaterData: () => TitleData[] = () =>
  require('./costaRicaUnderwater.json');
export const CostaRicaTopRatedData: () => TitleData[] = () =>
  require('./costaRicaTopRated.json');
export const AutoRotatorData: RotatorData[] = require('./rotatorData.json');
