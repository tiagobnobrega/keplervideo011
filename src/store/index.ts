import AsyncStorage from '@react-native-async-storage/async-storage';
import { configureStore } from '@reduxjs/toolkit';
import focusSlice from './focus/focusSlice';
import searchSlice from './search/searchSlice';
import settingsReducer, { SettingsState } from './settings/SettingsSlice';
import videoDetailReducer, {
  VideoDetailState,
} from './videoDetail/videoDetailSlice';

interface Store {
  videoDetail: VideoDetailState;
  settings: SettingsState;
}

const loadState = async () => {
  try {
    const serializedState = await AsyncStorage.getItem('reduxState');
    if (serializedState === null) {
      return undefined;
    }
    return JSON.parse(serializedState);
  } catch (err) {
    console.error('Could not load state', err);
    return undefined;
  }
};

const saveState = async (state: Store) => {
  try {
    const serializedState = JSON.stringify(state);
    await AsyncStorage.setItem('reduxState', serializedState);
  } catch (err) {
    console.error('Could not save state', err);
  }
};

export let store: any;

export const initializeStore = async () => {
  const preloadedState = await loadState();
  store = configureStore({
    reducer: {
      //@ts-ignore
      videoDetail: videoDetailReducer,
      settings: settingsReducer,
      search: searchSlice,
      focus: focusSlice,
    },
    preloadedState,
  });

  store.subscribe(() => {
    saveState(store.getState());
  });

  return store;
};

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
