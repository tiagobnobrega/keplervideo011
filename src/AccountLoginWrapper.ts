import {
  AccountLoginServerComponent,
  IAccountLoginHandlerAsync,
  IAccountLoginServerAsync,
  IStatus,
  StatusType,
} from '@amzn/kepler-media-account-login';
import { store } from './store';

const accountLoginServerComponent = new AccountLoginServerComponent();

export class AccountLoginWrapper {
  m_accountLoginServer?: IAccountLoginServerAsync;

  async updateStatus(loginStatus: boolean) {
    console.info(
      'AccountLoginWrapper::updateStatus invoked with: ',
      loginStatus,
    );
    try {
      const statusBuilder = accountLoginServerComponent.makeStatusBuilder();
      statusBuilder.status(
        loginStatus ? StatusType.SIGNED_IN : StatusType.SIGNED_OUT,
      );
      const status = statusBuilder.build();
      await this.m_accountLoginServer?.updateStatus(status);
    } catch (err) {
      console.error(
        'AccountLoginWrapper::updateStatus Failed updating login status: ',
        err,
      );
    }
  }

  setupAccountLoginServer() {
    try {
      this.m_accountLoginServer = accountLoginServerComponent.getOrMakeServer();
    } catch (err) {
      this.m_accountLoginServer = undefined;
      console.error(
        'AccountLoginWrapper::setupAccountLoginServer failed creating account login server: ',
        err,
      );
      return;
    }

    const accountLoginHandler: IAccountLoginHandlerAsync = {
      handleReadStatus(): Promise<IStatus> {
        console.info('AccountLoginWrapper::handleReadStatus invoked.');
        try {
          // Example of returning an app Login status from Redux
          const statusBuilder = accountLoginServerComponent.makeStatusBuilder();
          const appLoginStatus = store.getState().settings.loginStatus;

          statusBuilder.status(
            appLoginStatus ? StatusType.SIGNED_IN : StatusType.SIGNED_OUT,
          );

          const status = statusBuilder.build();

          console.info('AccountLoginWrapper:: App read status: ', status);

          return Promise.resolve(status);
        } catch (err) {
          console.error(
            'AccountLoginWrapper::handleReadStatus exception: ',
            err,
          );
        }

        return Promise.reject(
          'AccountLoginWrapper::handleReadStatus creating login status failed.',
        );
      },
    };

    try {
      console.info('AccountLoginWrapper::set Handler in process.');
      this.m_accountLoginServer?.setHandler(accountLoginHandler);
    } catch (err) {
      console.error(
        'AccountLoginWrapper::setupAccountLoginServer failed to set handler: ',
        err,
      );
    }
  }

  onStart(): Promise<void> {
    console.info('AccountLoginWrapper onStart()');
    this.setupAccountLoginServer();
    return Promise.resolve();
  }

  onStop(): Promise<void> {
    // add stop service code here.
    console.info('AccountLoginWrapper onStop()');
    return Promise.resolve();
  }
}

export const AccountLoginWrapperInstance = new AccountLoginWrapper();

export const onStartService = (): Promise<void> => {
  return AccountLoginWrapperInstance.onStart();
};

export const onStopService = (): Promise<void> => {
  return AccountLoginWrapperInstance.onStop();
};
