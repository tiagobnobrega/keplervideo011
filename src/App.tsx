/*
 * Copyright (c) 2022 Amazon.com, Inc. or its affiliates.  All rights reserved.
 *
 * PROPRIETARY/CONFIDENTIAL.  USE IS SUBJECT TO LICENSE TERMS.
 */
import { ThemeProvider } from '@amzn/kepler-ui-components';
import {
  useHideSplashScreenCallback,
  usePreventHideSplashScreen,
} from '@amzn/react-native-kepler';
import { NavigationContainer } from '@amzn/react-navigation__native';
import { Store } from '@reduxjs/toolkit';
import React, { useEffect, useMemo, useState } from 'react';
import { Provider } from 'react-redux';
import AppStack from './components/navigation/AppStack';
import { initializeStore } from './store';
import { createTheme } from './styles/ThemeBuilders';

const App = () => {
  usePreventHideSplashScreen();

  const [isAppReady, setIsAppReady] = useState(false);
  const [store, setStore] = useState<Store | null>(null);

  setTimeout(() => setIsAppReady(true), 4000);
  const hideSplashScreenCallback = useHideSplashScreenCallback();

  const initStore = async () => {
    const initializedStore = await initializeStore();
    setStore(initializedStore);
  };

  useEffect(() => {
    initStore();
  }, []);

  useEffect(() => {
    if (isAppReady) {
      // Hide the splash screen and render content
      hideSplashScreenCallback();
    }
  }, [isAppReady, hideSplashScreenCallback]);
  const theme = useMemo(() => createTheme(), []);

  if (!store) {
    return null;
  }

  return (
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <NavigationContainer>
          <AppStack />
        </NavigationContainer>
      </ThemeProvider>
    </Provider>
  );
};

export default App;
