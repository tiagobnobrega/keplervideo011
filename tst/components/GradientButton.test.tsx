import { render } from '@testing-library/react-native';
import React from 'react';
import 'react-native';
import GradientButton from '../../src/components/GradientButton';
import { L2_GRADIENT_COLORS } from '../../src/constants';

const onPressMock = jest.fn();
describe('GradientButton', () => {
  it('renders correctly', () => {
    const tree = render(
      <GradientButton
        label="Press Me"
        onPress={onPressMock}
        colors={L2_GRADIENT_COLORS}
        variant="secondary"
      />,
    );
    expect(tree).toMatchSnapshot();
  });
});
