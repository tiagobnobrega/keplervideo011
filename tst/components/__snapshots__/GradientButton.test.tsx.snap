// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GradientButton renders correctly 1`] = `
<GradientButton
  colors={
    Array [
      "#A3A3A399",
      "#5B5B5B99",
      "#3D3D3D",
    ]
  }
  label="Press Me"
  onPress={[MockFunction]}
  variant="secondary"
/>
`;
